File: time_float_backend/src/main/java/com/time_float_backend/entity/Reply.java
package com.time_float_backend.entity;

import java.time.LocalDateTime;

public class Reply {
    private Long id;
    private Long bottleId;
    private Long fromUserId;
    private Long toUserId;
    private String content;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private String fromUsername;
    private String toUsername;

    public Reply() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getBottleId() {
        return bottleId;
    }

    public void setBottleId(Long bottleId) {
        this.bottleId = bottleId;
    }

    public Long getFromUserId() {
        return fromUserId;
    }

    public void setFromUserId(Long fromUserId) {
        this.fromUserId = fromUserId;
    }

    public Long getToUserId() {
        return toUserId;
    }

    public void setToUserId(Long toUserId) {
        this.toUserId = toUserId;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    public String getFromUsername() {
        return fromUsername;
    }

    public void setFromUsername(String fromUsername) {
        this.fromUsername = fromUsername;
    }

    public String getToUsername() {
        return toUsername;
    }

    public void setToUsername(String toUsername) {
        this.toUsername = toUsername;
    }
}

File: time_float_frontend/lib/models/reply.dart
class Reply {
  final int id;
  final int bottleId;
  final int fromUserId;
  final int toUserId;
  final String content;
  final DateTime createdAt;
  final String? fromUsername;
  final String? toUsername;

  Reply({
    required this.id,
    required this.bottleId,
    required this.fromUserId,
    required this.toUserId,
    required this.content,
    required this.createdAt,
    this.fromUsername,
    this.toUsername,
  });

  factory Reply.fromJson(Map<String, dynamic> json) {
    return Reply(
      id: json['id'],
      bottleId: json['bottleId'],
      fromUserId: json['fromUserId'],
      toUserId: json['toUserId'],
      content: json['content'],
      createdAt: DateTime.parse(json['createdAt']),
      fromUsername: json['fromUsername'],
      toUsername: json['toUsername'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'bottleId': bottleId,
      'fromUserId': fromUserId,
      'toUserId': toUserId,
      'content': content,
      'createdAt': createdAt.toIso8601String(),
      'fromUsername': fromUsername,
      'toUsername': toUsername,
    };
  }
}
